libexec/bats-core/bats-preprocess

```diff
@@ -35,8 +35,8 @@ test_file="$1"
 tests=()
 {
   while IFS= read -r line; do
-    line="${line//$'\r'/}"
-    if [[ "$line" =~ $BATS_TEST_PATTERN ]]; then
+    line="${line//$'\r'}"
+    if [[ "$line" =~ $BATS_TEST_PATTERN ]] || [[ "$line" =~ $BATS_TEST_PATTERN_COMMENT ]]; then
```

Comment:
I wonder what the performance penalty is here (scratching up some perf regression tests I'll try this on).

If it's tangible, and there's a way of merging these regexs, that may help.

Reply:
I was unable to come up with a way to merge them (mighta been able to with PCRE, but posix regex seems to have too many limitations).  That is not to say someone smarter couldn't :).  That said, this check seems like it would be far less costly than running the tests themselves, so likely not a huge problem in any case.

Reply:
The two regexes could be merged via `MERGED_REGEX="($REGEX1)|($REGEX2)"`. However, that would lead to different capture groups beeing used for the name. The two regex parts would become capture groups so it would be easy to identify which one matched and go on from there. However, I also doubt that this would pose a significant performance problem right now, since the overhead per test is already at 100ms.